### PR TITLE
Some tools

### PR DESCRIPTION
**Pull request by ruanpepe**
I created a way to optionaly use a custom table to record translations to a specific table.
Changed points to use app default locale, and unlimited possible locales.

### CODE REVIEW

**Comment by chadidi** [comment]
Thanks, @ruanpepe for you contribution will review this PR ASAP!

**Comment by ruanpepe** [comment]
I think that some of these changes may be unwanted. I can undo them if necessary.

**Comment by chadidi** [comment]
Hey @ruanpep,

was going through your changes, unfortunately, can't merge all of the changes, as they are not related, which makes it hard for me to test it. please split these changes into small PR's

create a branch for each feature and open PR.

- [ ] feature/dynamic-translations-model
- [ ] deprecating package config in favor of Laravel locales config file
- [ ] other features/fixes

When opening the PR make sure your title describes the changes if possible, you can also use the description for more details.

Thanks 🙌

File: src/Translatable.php [hunk 1]
```diff
@@ -68,7 +68,7 @@ public function scopeWithTranslation(Builder $query, $locale = null, $fallback =
         }
 
         if ($fallback === true) {
-            $fallback = config('app.fallback_locale', 'en');
+            $fallback = config('app.fallback_locale');
```
Comment (chadidi) [inline src/Translatable.php:71]: need to keep the defaults in case this config key doesn't exist.

File: src/Translatable.php [hunk 2]
```diff
@@ -172,13 +168,21 @@ public static function scopeWhereTranslation($query, $field, $operator, $value =
...
         }
 
-        $response = [];
-        foreach ($languages as $language) {
-            $response[$language] = $this->getTranslatedAttribute($attribute, $language, $fallback);
+        $translations = $this->getRelation('translations')
+            ->where('column_name', $attribute);
+
+        if (!empty($languages)) {
+            $translations = $translations->whereIn('locale', $languages);
```
Comment (chadidi) [inline src/Translatable.php:181]: We should use this method `getTranslatedAttribute` here.

File: src/Translatable.php [hunk 3]
```diff
@@ -276,7 +278,17 @@ public function getTranslatableAttributes()
     }
 
     /**
-     * Set translations attributes.
+     * Get translations model.
+     *
+     * @return array
+     */
+    public function getTranslationsModel()
```
Comment (chadidi) [inline src/Translatable.php:285]: Test throwing this exception:
```
LogicException: LaravelArab\Tarjama\Tests\Models\Article::getTranslationsModel must return a relationship instance.
```

**Review by chadidi — CHANGES_REQUESTED** [review]

**Comment by chadidi** [comment]
Closing this as it was hard to make sure it didn't break anything without tests, now we have some basic tests but the PR didn't pass after the rebase on the master branch. So will just create an issue for each feature requested, if missing something open a new issue. Thanks

**Pull request closed without merging**